src/backend/distributed/metadata/node_metadata.c

```diff
@@ -753,13 +754,662 @@ PgDistTableMetadataSyncCommandList(void)
+		}
+	}
+
+	if (nodesWithMetadata == NIL)
+	{
+		return;
+	}
+
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 0;
+
+	Relation relation = table_open(DistPartitionRelationId(), AccessShareLock);
+
+	TupleDesc tupleDesc = RelationGetDescr(relation);
+	SysScanDesc scanDesc = systable_beginscan(relation,
+											  InvalidOid, false,
+											  NULL, scanKeyCount, scanKey);
+
+	MemoryContext metadataSyncContext = AllocSetContextCreate(CurrentMemoryContext,
+															  "metadata sync context",
+															  ALLOCSET_DEFAULT_SIZES);
+	MemoryContext oldContext = MemoryContextSwitchTo(metadataSyncContext);
+
+	bool scanEnded = false;
+	HeapTuple nextTuple = NULL;
+	while (!scanEnded)
+	{
+		MemoryContextReset(metadataSyncContext);
+
+		MemoryContextSwitchTo(oldContext);
+		nextTuple = systable_getnext(scanDesc);
+		if (!HeapTupleIsValid(nextTuple))
+		{
+			scanEnded = true;
+			MemoryContextSwitchTo(metadataSyncContext);
+			continue;
+		}
+		else if (!ShouldMetadataSyncHeapTuple(nextTuple, tupleDesc))
+		{
+			MemoryContextSwitchTo(metadataSyncContext);
+			continue;
+		}
+		MemoryContextSwitchTo(metadataSyncContext);
+
+		if (nextTuple)
+		{
+			bool isNullArray[Natts_pg_dist_partition];
+			Datum datumArray[Natts_pg_dist_partition];
+			heap_deform_tuple(nextTuple, tupleDesc, datumArray,
+							  isNullArray);
+			Datum relationIdDatum = datumArray[Anum_pg_dist_partition_logicalrelid - 1];
+			Oid relationId = DatumGetObjectId(relationIdDatum);
+
+			/*
+			 * skip foreign key and partition creation when the Citus table is
+			 * owned by an extension
+			 */
+			if (IsTableOwnedByExtension(relationId))
+			{
+				continue;
+			}
+
+			List *commandListForRelation =
+				InterTableRelationshipOfRelationCommandList(relationId);
+
+			commandListForRelation = lcons(DISABLE_DDL_PROPAGATION,
+										   commandListForRelation);
+			commandListForRelation = lappend(commandListForRelation,
+											 ENABLE_DDL_PROPAGATION);
+
+			SendMetadataCommandListToWorkerListInCoordinatedTransaction(nodesWithMetadata,
+																		CurrentUserName(),
+																		commandListForRelation);
+		}
+	}
+
+	systable_endscan(scanDesc);
+	table_close(relation, AccessShareLock);
+	list_free(nodesWithMetadata);
+
+	MemoryContextSwitchTo(oldContext);
+	MemoryContextDelete(metadataSyncContext);
+}
+
+
+/*
+ * SendIterableReplicationCommandsForDistObjects sends replication commands
+ * for distributed objects to workers within metadata connection.
+ */
+void
+SendIterableReplicationCommandsForDistObjects(List *nodeList)
+{
+	/* find relevant worker nodes */
+	List *nodesWithMetadata = NIL;
+	WorkerNode *workerNode = NULL;
+	foreach_ptr(workerNode, nodeList)
+	{
+		if (NodeIsPrimary(workerNode) && !NodeIsCoordinator(workerNode))
+		{
+			nodesWithMetadata = lappend(nodesWithMetadata, workerNode);
+		}
+	}
+
+	if (nodesWithMetadata == NIL)
+	{
+		return;
+	}
+
+	MemoryContext metadataSyncContext = AllocSetContextCreate(CurrentMemoryContext,
```

Comment:
for debugging purposes, should we give a specific context name for each? Like `metadata sync context:  distributed objects`